server/src/objects/player.ts

```diff
@@ -558,6 +719,82 @@ export class Player extends BaseGameObject {
         }
     }
 
+    useBandage(): void {
+        if (this.health == 100 || this.actionType == GameConfig.Action.UseItem) {
+            return;
+        }
+
+        // healing gets action priority over reloading
+        if (this.actionType == GameConfig.Action.Reload) {
+            this.cancelAction();
+            this.performActionXTick = true;
+            this.actionItemXTick = "bandage";
+            this.actionTypeXTick = GameConfig.Action.UseItem;
+            this.timeSincePerformActionXTick = Date.now();
+            return;
+        }
+
+        this.cancelAction();
+        this.doAction("bandage", GameConfig.Action.UseItem, 3);
+    }
+
+    useHealthkit(): void {
+        if (this.health == 100 || this.actionType == GameConfig.Action.UseItem) {
+            return;
+        }
+
+        // healing gets action priority over reloading
+        if (this.actionType == GameConfig.Action.Reload) {
+            this.cancelAction();
+            this.performActionXTick = true;
+            this.actionItemXTick = "healthkit";
+            this.actionTypeXTick = GameConfig.Action.UseItem;
+            this.timeSincePerformActionXTick = Date.now();
+            return;
+        }
+
+        this.cancelAction();
+        this.doAction("healthkit", GameConfig.Action.UseItem, 6);
+    }
+
+    useSoda(): void {
+        if (this.boost == 100 || this.actionType == GameConfig.Action.UseItem) {
```

Comment:
iirc surviv still allowed using sodas and pain killers if your adrenaline is full (which only really happened if you had the leadership perk)